plugins/BinarySearcher/DialogBinaryString.cpp

```diff
@@ -84,30 +85,35 @@ void DialogBinaryString::doFind() {
 			}
 
 			const size_t page_count      = region_size / page_size;
-			const QVector<uint8_t> pages = edb::v1::read_pages(region->start(), page_count);
-
-			if (!pages.isEmpty()) {
-
-				const uint8_t *p               = &pages[0];
-				const uint8_t *const pages_end = &pages[0] + region_size - sz;
-
-				while (p < pages_end) {
-					// compare values..
-					if (std::memcmp(p, b.constData(), sz) == 0) {
-						const edb::address_t addr  = p - &pages[0] + region->start();
-						const edb::address_t align = 1 << (ui.cmbAlignment->currentIndex() + 1);
-
-						if (!ui.chkAlignment->isChecked() || (addr % align) == 0) {
+			// Read out 4096 pages at a time, as some applications have huge regions
+			// and we will push against memory fragmentation if we mirror those regions.
+			// To prevent missing a needle in the haystack if it is split between read
+			// boundaries. increment current_page by only 4095.
```

Comment:
I guess you mean a comma instead of a period.

More to the point. What if the needle is longer than 4K? I think this algorithm will still miss it.

Reply:
Perhaps we can have it pick a size that is the size of the needle rounded up to nearest multiple of 4096?

Reply:
Do we actually expect anyone to use the binary search feature on a string larger than 4K pages? At a minimum this is 4096 bytes if the page size was one, but realistically we're looking at 4KiB pages meaning we're not bugged unless we exceed 16MiB.

Reply:
> Do we actually expect anyone to use the binary search feature on a string larger than 4K pages? At a minimum this is 4096 bytes if the page size was one, but realistically we're looking at 4KiB pages meaning we're not bugged unless we exceed 16MiB.

Actually, it was my mistake: I though about 4K bytes rather than pages. Indeed, it should have been 4K pages, which wouldn't make sense for the entry fields in the dialog. Maybe all that needed to be done was setting a limit on the length of these fields (`maxLength`), just in case someone pastes such a large piece of data.

Reply:
@10110111, so do you think it's ready to merge then?

Reply:
Yep.